source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/filter.cc

```diff
@@ -95,44 +97,63 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
         Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(decoder_callbacks_);
 
     if (per_route_config != nullptr && per_route_config->disabled()) {
-      ENVOY_STREAM_LOG(debug,
-                       "Transcoding is disabled for the route. Request headers are passed through.",
-                       *decoder_callbacks_);
+      ENVOY_LOG(debug,
+                "Transcoding is disabled for the route. Request headers are passed through.");
 
       enabled_ = false;
       return Http::FilterHeadersStatus::Continue;
     }
   }
 
   if (Envoy::Grpc::Common::isGrpcRequestHeaders(headers)) {
-    auto method = static_cast<std::string>(headers.getPathValue());
-    auto status = transcoder_.prepareTranscoding(method);
 
-    if (!status.ok()) {
-      ENVOY_STREAM_LOG(error, "Failed to prepare transcoding. Abort Processing. Error was: {}",
-                       *decoder_callbacks_, status.message());
+    // Configure transcoder to process this request.
+    auto status_method = httpMethodFrom(headers.getMethodValue());
+    if (!status_method.ok()) {
+      ENVOY_LOG(error, "Failed to parse HTTP Method. Abort Processing. Error was: {}",
+                status_method.status().message());
+
+      respondWithGrpcError(*decoder_callbacks_, Errors::UnexpectedMethodType);
+      return Http::FilterHeadersStatus::StopIteration;
+    }
+
+    auto path = static_cast<std::string>(headers.getPathValue());
+    auto method_and_path = HttpMethodAndPath{*status_method, std::move(path)};
```

Comment:
Is status_metod string? Can we move it as well?

Reply:
Nope its an enum.